retry/retry.go

```diff
@@ -54,11 +81,10 @@ func (r Retryer) Do(ctx context.Context, method Retryable) error {
 		timer := time.NewTimer(r.waitDuration)
 		select {
 		case <-timer.C:
+		case <-timeoutCtx.Done():
+			return timeoutCtx.Err()
```

Comment:
You should specialize the context errors here.

Reply:
Done